City-of-Helsinki/drupal-helfi-platform-config
# UHF-4528: Large amount of units listing

# Add a way to list large amounts of units on a listing [UHF-4528](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4528) & [UHF-4532](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4532) 
Make it possible for the content producers to tag and list large numbers of units at the same time (for example daycare units).

## What was done
* Added views bulk operations and views bulk edit modules.
* Added Unit type vocabulary for the categorisation.
* Added new field Unit type for TPR-units.
* Changed the TPR-unit listing so that you can do views bulk edit on it.
* Added minimal display for TPR units.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config and hdbt
    * `composer require drupal/helfi_platform_config:dev-UHF-4528_large_amount_of_units_listing -W && composer require drupal/hdbt:dev-UHF-4528_large_amount_of_units_listing -W`
* Run `make drush-updb drush-cr`

## How to test
* Go to `/admin/structure/taxonomy/manage/unit_type/overview` and add one term there for example _Daycare_
* Go to `/admin/content/integrations/tpr-unit` listing and check that there is action called _Modify values_ on the bottom of the listing. Select some units and do the action _Modify values_ for those units. Pick the term you just created for the _Unit type_ field on the selected units and save.
* Go to check any of the units that you edited if there is the term set on the _Unit type_ field.

## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/270
* https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/116
